lib/index.ts

```diff
@@ -0,0 +1,405 @@
+   */
+  constructor(options: string | ClientOptions) {
+    if (typeof options === "string") {
+      this.token = options;
+      this.BASE_URL = "https://api.topstats.gg";
+    } else {
+      this.token = options.token;
+      this.BASE_URL = options.baseUrl ?? "https://api.topstats.gg";
+    }
+  }
+
+  /**
+   * Validates a Discord bot ID format
+   * @throws {TopStatsError} If ID format is invalid
+   */
+  private validateBotId(botId: string): void {
+    if (!/^\d{17,19}$/.test(botId)) {
+      throw new TopStatsError("Invalid Discord bot ID format");
+    }
+  }
+
+  /**
+   * Makes an HTTP request to the API
+   * @throws {RateLimitError} When rate limited
+   * @throws {TopStatsError} For other API errors
+   */
+  private async _request<T>(
+    method: string,
+    path: string,
+    query?: Record<string, any>,
+  ): Promise<T> {
+    try {
+      const url = new URL(`${this.BASE_URL}${path}`);
+
+      if (query && method === "GET") {
+        Object.entries(query).forEach(([key, value]) => {
+          if (value !== undefined) {
+            url.searchParams.append(key, String(value));
+          }
+        });
+      }
+
+      const response = await fetch(url.toString(), {
+        method,
+        headers: {
+          Authorization: this.token,
+          "Content-Type": "application/json",
+          "User-Agent": "TopStats/1.0.0",
+        },
+        body: method !== "GET" && query ? JSON.stringify(query) : undefined,
+      });
+
+      const data = await response.json();
+
+      if (response.status === 429) {
+        const rateLimitData = data as RateLimitedResponse;
+        throw new RateLimitError(
+          rateLimitData.message,
+          rateLimitData.expiresIn,
+        );
+      }
+
+      if (!response.ok) {
+        throw new TopStatsError(
+          `API Error ${response.status}: ${response.statusText}`,
+        );
+      }
+
+      return data as T;
+    } catch (error) {
+      if (error instanceof TopStatsError) throw error;
+      throw new TopStatsError(
+        error instanceof Error ? error.message : "Unknown error occurred",
+      );
+    }
+  }
+
+  /**
+   * Fetches detailed information about a specific bot
+   * @param botId - The Discord bot ID
+   * @throws {TopStatsError} If bot ID is invalid
+   */
+  async getBot(botId: string): Promise<BotData> {
+    this.validateBotId(botId);
+    return this._request<BotData>("GET", `/discord/bots/${botId}`);
+  }
+
+  /**
+   * Fetch historical data for a specific bot
+   * @param botId - The Discord bot ID
+   * @param timeFrame - Time frame for historical data
+   * @param type - Type of data to fetch
+   */
+  async getBotHistorical<T extends HistoricalDataTypes>(
+    botId: string,
+    timeFrame: HistoricalDataTimeFrame,
+    type: T,
+  ): Promise<{
+    data: Array<Extract<HistoricalData, { type: T }>>;
+  }> {
+    this.validateBotId(botId);
+    return this._request<{
+      data: Array<Extract<HistoricalData, { type: T }>>;
```

Comment:
If the type string of the HistoricalData's is going to be used like this it should be an enum

Reply:
fixed